### PR TITLE
[miio] Allow for quantity Type

### PR DESCRIPTION
**Pull request by marcelrv**
e.g. use number:temperature

readonly temperature tested.
note: needs further testing on dimensionless and handle command

Signed-off-by: Marcel Verpaalen <marcel@verpaalen.com>

### CODE REVIEW

**Comment by marcelrv** [comment]
@cpmeister do you know if there is a way to get the  unit, e.g. SIUnits.CELSIUS from the string "Celsius' or e.g.  SmartHomeUnits.AMPERE from the string "ampere"  I don't seem to find a 'Unit.getUnit(String)' sort of function.

In the end, I'm  want to  create a QuantityType<?> with the right format having the string "Celsius"... (note I found .toUnit but that takes °C to convert instead of celsius.)

Alternatively I can make my own enum to convert the string, but that feels like the wrong approach

**Comment by cpmeister** [comment]
I tried looking for something that you could use but it seems that you are right, there is no easy utility in the core or the units library that would return a Units instance based off of the formal name of the unit. The primary reason for this is that the formal names of the units only exist as class field names. The only existing lookup method that is close to what you are looking for is `UnitUtils.parseUnit(String)`. But even that method is expecting the unit symbol, not the unit name. 

If you really need a way to create a unit based off of the unit name you would have to create it yourself right now. You can either create a mapping function or use reflection to lookup unit fields based off their field name.

**Comment by marcelrv** [comment]
thanks @cpmeister for the response on the conversion. I thought I was crazy not to find such thing.
I indeed created my own little conversion, with the units I've seen so far. Works well in my testing and indeed does the automatic conversions for e.g. folks having imperial units.

File: bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoBasicHandler.java [hunk 1]
```diff
@@ -534,6 +562,37 @@ private void updateChannel(@Nullable MiIoBasicChannel basicChannel, String param
...
+            case "electriccurrent":
+                updateState(basicChannel.getChannel(),
+                        new QuantityType<>(val.getAsBigDecimal(), SmartHomeUnits.AMPERE));
+                break;
+            case "energy":
+                updateState(basicChannel.getChannel(), new QuantityType<>(val.getAsBigDecimal(), SmartHomeUnits.WATT));
+            case "time":
+                updateState(basicChannel.getChannel(), new QuantityType<>(val.getAsBigDecimal(), SmartHomeUnits.HOUR));
+            default:
+                updateState(basicChannel.getChannel(), new DecimalType(val.getAsBigDecimal()));
```
Comment (cpmeister) [inline bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoBasicHandler.java:592]: you forgot to add break statements.

**Review by cpmeister — CHANGES_REQUESTED** [review]
These changes look like a great start. Are you going to adjust more channels or are you just doing temperature for now?

**Comment by marcelrv** [comment]
This PR will be basically just looking at adding the functionality.
I still need to experiment a bit how OH is behaving if you have already your item definitions and you are changing it to a quantity type. (e.g what happens if you keep your definition files the old ones, but the thing expects a quanity type.)

If there is no adverse behavior, I'll start changing the json db files.

In the next few weeks I'm planning to add a next wave of new devices, those I'll create with these new quantity types from the gekko.

**Review by cpmeister — APPROVED** [review]
If you feel this is ready to merge let me know.

**Comment by marcelrv** [comment]
thanks, would be great if it can be merged.

**Pull request merged**